File: ui/src/app/pages/map/zoom-level.ts
export class ZoomLevel {

  static readonly bitmapTileMinZoom = 6;
  static readonly bitmapTileMaxZoom = 11;

  static readonly vectorTileMinZoom = 12;
  static readonly vectorTileMaxZoom = 15;
  static readonly vectorTileMaxOverZoom = 20;
  static readonly vectorTileFullDetailZoom = 14;

  static readonly minZoom: number = Math.min(ZoomLevel.bitmapTileMinZoom, ZoomLevel.vectorTileMinZoom);
  static readonly maxZoom: number = Math.min(ZoomLevel.bitmapTileMaxZoom, ZoomLevel.vectorTileMaxZoom);

}

File: production/shared/src/main/scala/kpn/shared/tiles/ZoomLevel.scala
package kpn.shared.tiles

import scala.math.min

object ZoomLevel {

  val bitmapTileMinZoom = 6
  val bitmapTileMaxZoom = 11

  val vectorTileMinZoom = 12
  val vectorTileMaxZoom = 15
  val vectorTileMaxOverZoom = 20
  val vectorTileFullDetailZoom = 14

  val minZoom: Int = min(bitmapTileMinZoom, vectorTileMinZoom)
  val maxZoom: Int = min(bitmapTileMaxZoom, vectorTileMaxZoom)

}
